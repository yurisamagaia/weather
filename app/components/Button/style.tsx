import styled from 'styled-components/native';
import { colors } from '../../theme';

export const Container = styled.TouchableOpacity`
  width: 150px;
  background-color: ${colors.white};
  border-radius: 30px;
  justify-content: center;
  align-items: center;
  margin: 5px;
  padding: 10px;
  border: 1px solid ${colors.blueDark};
`;

export const Text = styled.Text`
  color: ${colors.blue};
  font-size: 20px;
  font-weight: bold;
`;