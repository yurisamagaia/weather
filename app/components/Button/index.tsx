import React, { FC } from 'react';
import { Container, Text } from './style';

type Props = {
  onPress: () => void
};

const Button: FC<Props> = ({ children, onPress }) => (
  <Container onPress={onPress}>
    <Text>{children}</Text>
  </Container>
);

export default Button;