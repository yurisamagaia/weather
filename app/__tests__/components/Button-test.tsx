import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../components/Button';

describe('Button tests', () => {
  // it('Should render correctly', async () => {
  //   const tree = renderer.create(<Button />).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  it('Simulator onClick instead of onPress', async () => {
    const callback = jest.fn();
    const component = renderer.create(<Button onPress={callback} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});