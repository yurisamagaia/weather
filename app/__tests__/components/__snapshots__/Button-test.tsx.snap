// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests Simulator onClick instead of onPress 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderColor": "#009ACD",
      "borderRadius": 30,
      "borderStyle": "solid",
      "borderWidth": 1,
      "justifyContent": "center",
      "marginBottom": 5,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 5,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": 150,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#00BFFF",
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  />
</View>
`;
